src/processor/py/mod.rs

```diff
@@ -0,0 +1,1091 @@
+        let key = Arc::new(Mutex::new(key.to_string()));
+        let value = AnyValue {
+            value: Arc::new(Mutex::new(Some(BoolValue(b)))),
+        };
+        let value = Arc::new(Mutex::new(Some(value)));
+        Ok(PyKeyValue {
+            inner: Arc::new(Mutex::new(KeyValue { key, value })),
+        })
+    }
+    // Helper methods for class
+    #[staticmethod]
+    fn new_int_value(key: &str, py: Python, value: PyObject) -> PyResult<PyKeyValue> {
+        let i = value.extract::<i64>(py)?;
+        let key = Arc::new(Mutex::new(key.to_string()));
+        let value = AnyValue {
+            value: Arc::new(Mutex::new(Some(IntValue(i)))),
+        };
+        let value = Arc::new(Mutex::new(Some(value)));
+        Ok(PyKeyValue {
+            inner: Arc::new(Mutex::new(KeyValue { key, value })),
+        })
+    }
+    // Helper methods for class
+    #[staticmethod]
+    fn new_double_value(key: &str, py: Python, value: PyObject) -> PyResult<PyKeyValue> {
+        let f = value.extract::<f64>(py)?;
+        let key = Arc::new(Mutex::new(key.to_string()));
+        let value = AnyValue {
+            value: Arc::new(Mutex::new(Some(DoubleValue(f)))),
+        };
+        let value = Arc::new(Mutex::new(Some(value)));
+        Ok(PyKeyValue {
+            inner: Arc::new(Mutex::new(KeyValue { key, value })),
+        })
+    }
+    #[getter]
+    #[allow(deprecated)]
+    fn key<'py>(&self, py: Python<'py>) -> PyResult<PyObject> {
+        let v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        let binding = v.key.clone();
+        let bind_lock = binding.lock();
+        let x = Ok(bind_lock.unwrap().clone().into_py(py));
+        x
+    }
+    #[setter]
+    fn set_key(&mut self, new_value: &str) -> PyResult<()> {
+        let v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        let binding = v.key.clone();
+        let mut bind_lock = binding.lock().unwrap();
+        bind_lock.clear();
+        bind_lock.insert_str(0, new_value);
+        Ok(())
+    }
+    #[getter]
+    fn value(&self) -> PyResult<PyAnyValue> {
+        let v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        let binding = v.value.clone();
+        Ok(PyAnyValue {
+            inner: binding.clone(),
+        })
+    }
+    #[setter]
+    fn set_value(&mut self, new_value: &PyAnyValue) -> PyResult<()> {
+        let v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        let binding = v.value.clone();
+        let bind_lock = binding.lock();
+        let binding = new_value.inner.clone();
+        let x = binding.lock().unwrap();
+        bind_lock.unwrap().replace(x.clone().unwrap());
```

Comment:
Could these locks fail similar to the inner.lock that's checked above?

Reply:
Yeah per the first comment, need a pass through all the locking logic with tests as we familiarize with pyo3. Will add a linear address.